src/Application/Actions/RegularGivingMandate/Get.php

```diff
@@ -36,17 +37,22 @@ protected function action(Request $request, Response $response, array $args): Re
         if (! $this->environment->isFeatureEnabledRegularGiving()) {
             throw new HttpNotFoundException($request);
         }
+
+        Assertion::keyExists($args, "mandateId");
+        $mandateId = $args["mandateId"];
+        Assertion::string($mandateId);
         if (empty($args['mandateId'])) {
-            throw new DomainRecordNotFoundException('Missing donation ID');
+            throw new DomainRecordNotFoundException('Missing mandate ID ' . $mandateId);
         }
 
+        // @todo-regular-giving: make sure that the donor requesting the mandate is the owner of the mandate
```

Comment:
I think since this is a confidentiality requirement we should do this before we merge this PR to make sure it can't be forgotten.